metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricAnomalyDetectorService.java

```diff
@@ -0,0 +1,56 @@
+package org.hypertrace.alert.engine.metric.anomaly.detector;
+
+import java.io.IOException;
+import java.time.Duration;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.AlertTask;
+import org.hypertrace.alert.engine.metric.anomaly.datamodel.queue.KafkaAlertTaskConsumer;
+import org.hypertrace.core.serviceframework.PlatformService;
+import org.hypertrace.core.serviceframework.config.ConfigClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MetricAnomalyDetectorService extends PlatformService {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(MetricAnomalyDetectorService.class);
+
+  KafkaAlertTaskConsumer alertTaskConsumer;
+  MetricAnomalyDetector metricAnomalyDetector;
+
+  public MetricAnomalyDetectorService(ConfigClient configClient) {
+    super(configClient);
+  }
+
+  @Override
+  protected void doInit() {
+    alertTaskConsumer = new KafkaAlertTaskConsumer(getAppConfig().getConfig("queue.config.kafka"));
+    metricAnomalyDetector = new MetricAnomalyDetector(getAppConfig());
+  }
+
+  @Override
+  protected void doStart() {
+    while (true) {
+      try {
+        AlertTask alertTask = alertTaskConsumer.dequeue();
+        if (alertTask != null) metricAnomalyDetector.process(alertTask);
+      } catch (IOException e) {
+        LOGGER.error("Exception processing record", e);
+      }
+
+      try {
```

Comment:
curious: why are we adding a fixed 1 second delay?

Reply:
you mean it should be made configurable?

Reply:
Do we need it? I think, as soon as one task is done, it should pick up next, right?